### DISCUSSION TITLE
Docs: [Flat Config] Using with CLI

### DISCUSSION THREAD

**Discussion by saturdaywalkers** [eslint/eslint#17868] [Q&A]
### Docs page(s)

Flat Config Blog Posts - don't explain how to use flat config with the CLI

Its unclear why the flat config is different when using the CLI

Using eslint 5.86.0, node 20.5.1

### What documentation issue do you want to solve?

Many of the examples don't work or causes errors with CLI
`npx eslint js`

1) Have to start with 'module.exports`, but this is undocumented, e.g.

```
module.exports = [
	'eslint:all',
	{
	files					: [ '**/*.js' ],
        },
```

Both 'import' and 'export' cause errors.

2) There is an error message `" Warning: The 'eslint:all' string configuration is deprecated and will be replaced by the @eslint/js package's 'all' config."`, but import'ing that does not work with CLI + flat config

3) How do you configure `globals.browser` as import'ing does not work with CLI + flat config

### What do you think is the correct solution?

I'm not sure if this is a documentation issue, or a bug in eslint CLI 

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by saturdaywalkers** [comment]
ok, adding  `"type": "module"` to `package.json` meant I could use the flat config as described
this should be mandatory as things like` import globals from globals` wont work otherwise 

another gotcha, not yours but worth mentioning, was 
`TypeError: Global "AudioWorkletGlobalScope " has leading or trailing whitespace.
at Object.validate ( [deleted] /node_modules/eslint/lib/config/flat-config-schema.js:313:23)

had to `npm install globals` to fix it

Takeaway - need to add  `"type": "module"` to your `package.json`

**Comment by aladdin-add** [comment]
if you are using esm(`"type': "module"`), the `eslint.config.js` was treated as es module, so you can use `import/export`.
otherwise it was commonjs, you can use `require/modue.exports`.

> `TypeError: Global "AudioWorkletGlobalScope " has leading or trailing whitespace.

it was an issue in globals, but has been fixed, you'll need to upgrade globals to the latest.
https://github.com/sindresorhus/globals/pull/184
